superset/assets/src/visualizations/nvd3/NVD3Vis.js

```diff
@@ -320,25 +319,37 @@ function nvd3Vis(element, props) {
           if (tableFilter) {
             const publishedColumns = formData.publishColumns;
             let yColumn;
+            let yColumnLabel;
             let metric;
+            let key = e.series.key;
+
             if (formData.metrics.length == 1) {
               metric = formData.metrics[0];
             } else {
-              let key = e.series.key;
               metric = formData.metrics.find((metric) => {
-                return key.includes(metric.label);
+                return metric['expressionType'] ?  key.includes(metric.label) : key.includes(metric);
               })
             }
-            if (metric && metric.column) {
+
+            if (metric && metric.expressionType === undefined) {
+              yColumn = metric;
+              yColumnLabel = metric;
+            } else if (metric && metric.expressionType === 'SIMPLE') {
               yColumn = metric.column.column_name;
```

Comment:
will we not need `metric && metric.column` check for count(*) usecase u raised earlier

Reply:
no, that will fall under `metric.expressionType === 'SIMPLE'` case